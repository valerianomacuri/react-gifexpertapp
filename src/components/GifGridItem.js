import React from 'react'

export const GrifGridItem = ({id, title, url}) => {

    return (
        <div className="card animate_animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title.toUpperCase() }</p>
        </div>
    )
}