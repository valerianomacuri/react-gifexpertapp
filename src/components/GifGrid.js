import React from 'react'
import PropTypes from 'prop-types'

import { GrifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category )
    
    return (
        <>
            <h3 className="animate_animated animate__fadeIn">{ category }</h3>

            {loading && <p className="animate_animated animate__flash">Loading</p> }

            <div className="card-grid">
                { images.map( img => (
                    <GrifGridItem
                    key={img.id} 
                    {...img } />
                ) )}
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}