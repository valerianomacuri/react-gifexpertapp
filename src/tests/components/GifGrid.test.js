import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('<GifGrid />', () => {
    
    const category = 'One Punch'

    test('debe de mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={ category }/>)
        expect( wrapper ).toMatchSnapshot()

    });

    test('Debe de mostrar item cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                title: 'Cualquier cosa',
                url: 'https://localhost/cualquier/cosa.jpg'
            },
            {
                id: '123',
                title: 'Cualquier cosa',
                url: 'https://localhost/cualquier/cosa.jpg'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={ category }/>)
        // expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe( false )

        expect( wrapper.find('GrifGridItem').length ).toBe( gifs.length )

    });
    
    
});
