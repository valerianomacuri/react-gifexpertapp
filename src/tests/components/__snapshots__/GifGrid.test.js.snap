// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate_animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate_animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
